pkg/ingester/ingester.go

```diff
@@ -3137,10 +3151,42 @@ func (i *Ingester) Push(ctx context.Context, req *mimirpb.WriteRequest) (*mimirp
 		mimirpb.ReuseSlice(req.Timeseries)
 	})
 	err := i.PushWithCleanup(ctx, pushReq)
-	if err != nil {
-		return nil, err
+	if err == nil {
+		return &mimirpb.WriteResponse{}, nil
+	}
+	handledErr := handlePushError(err)
+	return nil, handledErr
+}
+
+func handlePushError(err error) error {
+	var (
+		badDataErr badDataError
+	)
+	switch {
+	case errors.As(err, &badDataErr):
+		return newErrorWithHTTPStatus(
+			err,
+			http.StatusBadRequest,
+		)
+	case errors.As(err, &unavailableError{}):
+		return newErrorWithStatus(
+			err,
+			codes.Unavailable,
+		)
+	case errors.As(err, &instanceLimitReachedError{}):
+		return newErrorWithStatus(
```

Comment:
nit: it would make tiny bit more sense to me to switch the wrapping here:

Ie. `util_log.DoNotLogError{Err: newErrorWithStatus(err, codes.Unavailable)}`

In practice, it doesn't matter.

Reply:
If you don't have a strong opinion on this, I would keep it as it is.